File: src/main/java/de/thatsich/intellie/common/module/AName.java
package de.thatsich.intellie.common.module;

import de.thatsich.intellie.common.BaseModID;

import java.beans.Introspector;

/**
 @author thatsIch
 @since 02.04.2014. */
public abstract class AName implements IName
{
	private final String unlocalizedName;

	protected AName( BaseModID id )
	{
		final String name = this.getClass().getSimpleName().replaceFirst( "Name", "" );

		this.unlocalizedName = Introspector.decapitalize( name );
	}

	@Override
	public String getUnlocalizedName() { return this.unlocalizedName; }
}

File: src/main/scala/de/thatsich/minecraft/intellie/common/module/AName.scala
package de.thatsich.minecraft.intellie.common.module

/**
 * Basic wrapper for a name
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class AName(val name: String)
{}

object AName
{
	implicit def nameToString(name: AName): String = name.name
}